### DISCUSSION TITLE
kit's version of node (15) is overriding my own version from n

### DISCUSSION THREAD

**Discussion by kentcdodds** [johnlindquist/kit#432] [Errors]
No matter what I do, my `node` resolves to one installed by `kit` which is causing me a bunch of issues because kit's version is 15...

```
~/code/remix-kentcdodds (main) 🧞‍
$ node --version
v15.13.0
~/code/remix-kentcdodds (main) 🧞‍
$ n
   installed : v16.9.1 to /Users/kentcdodds/n/bin/node
      active : v15.13.0 at /Users/kentcdodds/.kit/node/bin/node
~/code/remix-kentcdodds (main) 🧞‍
$ which node
/Users/kentcdodds/.kit/node/bin/node
~/code/remix-kentcdodds (main) 🧞‍
$ 
```

I use https://npm.im/n to manage my node version and even after selecting v16.9.1, it's still using node 15 from kit. I've closed and opened the terminal and everything. I can't get it to use the version from `n`. :(

**Comment by johnlindquist** [comment]
🤔

This might be an issue from an older Kit install.

Can you paste the result of `echo $PATH | pbcopy`? 

Kit shouldn't have affected your $PATH (which controls the order the terminal searches for commands), but maybe some old config did something we need clean up.

**Reply by kentcdodds** [reply]
Here we are:

```
/Users/kentcdodds/.yarn/bin:/Users/kentcdodds/.config/yarn/global/node_modules/.bin:/usr/local/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/kentcdodds/.kit/node/bin:/Users/kentcdodds/.kit/bin:./node_modules/.bin:/Users/kentcdodds/.bin:/Users/kentcdodds/.my_bin:/Users/kentcdodds/.kenv/bin:/Users/kentcdodds/.kit/bin:/Users/kentcdodds/n/bin
```

**Reply by johnlindquist** [reply]
Cool, thx.

We gotta get rid of that `Users/kentcdodds/.kit/node/bin:`

Can you check your ~/.zshrc (or bash/fish/whatever she'll rc you're using) for a line that looks like it adds that?

(This will have no impact on Kit)

**Reply by kentcdodds** [reply]
Yeah, I'm not sure where that's coming from. It's not in my `~/.zshrc` 🤔

**Reply by kentcdodds** [reply]
I just cleared out my `~/.zshrc` completely and still got:

```
/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/kentcdodds/.kit/node/bin:/Users/kentcdodds/.kit/bin
```

**Reply by johnlindquist** [reply]
Try `grep -R "kit/node" ~` and see if anything pops up

**Reply by kentcdodds** [reply]
I rebooted and now it's gone. At the top of my `~/.zshrc` file, my PATH now looks like:

```
/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin
```

So I'm not sure how this happened, but yeah, it seems to have fixed itself. *sigh*

**Reply by kentcdodds** [reply]
Ugh... It's back...

**Reply by kentcdodds** [reply]
Interestingly, it looks like now it only applies in the VSCode terminal 🤔

**Reply by kentcdodds** [reply]
Trying the grep thing. It's taking forever 🙃

**Reply by johnlindquist** [reply]
Darn, please ping me if it pops up again.

The only ways for $PATH to be changed are:
1. The shell setup configs: ~/.zshrc, etc [Article on config loading](https://medium.com/@rajsek/zsh-bash-startup-files-loading-order-bashrc-zshrc-etc-e30045652f2e)
2. From the shell, we `source foo.sh` and the contents of `foo.sh` contain a `PATH="$PATH:/some/other/dir"` (but that would only live for the current session)

My only other thought is if you're using the Terminal in VS Code. The "inherit env" setting has caused me troubles in the past:
![CleanShot 2021-09-13 at 13 18 54](https://user-images.githubusercontent.com/36073/133143611-64a99f7b-8505-4583-bbf6-e30a3057d054.png)

**Reply by johnlindquist** [reply]
Found my old twitter thread on the vs code env vars topic: https://twitter.com/tyriar/status/1344671459466125312

**Reply by kentcdodds** [reply]
Oh! This might actually be it because I launch VSCode from kit so if kit changes the env, then VSCode will inherit that.

I just verified this by launching vscode via Alfred and the issue no longer exists.

Unfortunately, unchecking that box doesn't appear to change anything :(

**Reply by johnlindquist** [reply]
Aha! Hmm, good catch. I'll get that sorted.

A "quick fix" might be:

`export PATH="Users/kentcdodds/n/bin:$PATH"` in your vs code terminal. Then n/bin should take priority in any future process.

**Reply by johnlindquist** [reply]
Also, just to confirm, "Launch code from Kit" means you're using the `edit()`  function in a Kit script?

**Reply by kentcdodds** [reply]
The grep finally finished and the only interesting thing that came out of it was this:

```
/Users/kentcdodds/Library/Containers/com.docker.docker/Data/log/host/com.docker.supervisor.log.2
2078:time="2021-08-12T09:29:43-06:00" level=info msg="pidfile \"com.docker.vpnkit\" references Pid 630 but paths differ: assuming pid re-use, removing it\n   expected: \"/Applications/Docker.app/Contents/Resources/bin/com.docker.vpnkit\"\n  effective: \"/Users/kentcdodds/.kit/node/bin/node\"\n"
```

Not sure whether that's related, and I'm not even sure what it's doing but I thought it was interesting.

**Reply by kentcdodds** [reply]
> Also, just to confirm, "Launch code from Kit" means you're using the `edit()` function in a Kit script?

Yes

**Reply by johnlindquist** [reply]
`edit` does unnecessarily load `.kit/node/bin` onto the PATH*, so it seems like that's the culprit

I see it happening in my VS Code env, but in my PATH my node is installed at `/usr/local/bin` which precedes `/Users/johnlindquist/.kit/node/bin` so I didn't notice.

*It's sharing PATH with the `await npm()` util which requires `.kit/node/bin` on the PATH, so I'll have to separate them.
